presets/ragengine/vector_store_manager/manager.py

```diff
@@ -10,9 +10,10 @@ class VectorStoreManager:
     def __init__(self, vector_store: BaseVectorStore):
         self.vector_store = vector_store
 
-    async def index(self, index_name: str, documents: List[Document]) -> List[str]:
+    async def index(self, index_name: str, documents: List[Document]) -> List[Dict[str, Any]]:
         """Index new documents."""
-        return await self.vector_store.index_documents(index_name, documents)
+        ids = await self.vector_store.index_documents(index_name, documents)
+        return await self.vector_store._get_index_documents_by_ids(index_name, ids)
```

Comment:
This addition fetches docs from the store based on ID's, which should be keyed for O(1) lookup time but will add some latency